chain/cosmos/module_feegrant.go

```diff
@@ -0,0 +1,66 @@
+package cosmos
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"cosmossdk.io/x/feegrant"
+)
+
+// FeeGrant grants a fee grant.
+func (tn *ChainNode) FeeGrant(ctx context.Context, granterKey, grantee, spendLimit string, allowedMsgs []string, expiration time.Time, extraFlags ...string) error {
+	cmd := []string{"feegrant", "grant", granterKey, grantee, "--spend-limit", spendLimit}
+
+	if len(allowedMsgs) > 0 {
+		msgs := make([]string, len(allowedMsgs))
+		for i, msg := range allowedMsgs {
+			if !strings.HasPrefix(msg, "/") {
+				msg = "/" + msg
+			}
+			msgs[i] = msg
+		}
+
+		cmd = append(cmd, "--allowed-messages", strings.Join(msgs, ","))
+	}
```

Comment:
out of curiosity, what happens if the `--allowed-messages` flag isn't used? does the granter give the grantee permission to execute all message types?

Reply:
correct, then it becomes a wildcard feegrant. (Just pay what ever fees I request of you within my fee limit)